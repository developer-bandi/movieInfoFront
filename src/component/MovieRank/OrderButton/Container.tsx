import { useDispatch, useSelector } from 'react-redux';
import { ReducerType } from '../../../store';
import { setRankType } from '../../../store/movieRank/Reducer';
import OrderButtonPresentational from './Presentational';

const OrderButtonContainer = () => {
  const orderType = useSelector(
    (state: ReducerType) => state.movieRank.content.type
  );
  const dispatch = useDispatch();
  const orderchange = (Type: string) => {
    dispatch(setRankType(Type));
  };
  return (
    <OrderButtonPresentational
      orderchange={orderchange}
      orderType={orderType}
    />
  );
};

export default OrderButtonContainer;
