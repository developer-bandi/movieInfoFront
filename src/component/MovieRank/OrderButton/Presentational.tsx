import styles from './Style';

interface OrderButtonProps {
  orderchange: (Type: string) => void;
  orderType: string;
}

const OrderButtonPresentational = ({
  orderType,
  orderchange,
}: OrderButtonProps) => {
  return (
    <styles.OrderButtonBlock>
      <styles.OrderButtonSubBlock>
        <styles.OrderButton
          data-testid={'popularOrderButton'}
          onClick={() => orderchange('popular')}
          active={orderType === 'popular'}
        >
          인기순
        </styles.OrderButton>
        <styles.OrderButton
          data-testid={'topRatedOrderButton'}
          onClick={() => orderchange('topRated')}
          active={orderType === 'topRated'}
        >
          별점순
        </styles.OrderButton>
      </styles.OrderButtonSubBlock>
    </styles.OrderButtonBlock>
  );
};

export default OrderButtonPresentational;
