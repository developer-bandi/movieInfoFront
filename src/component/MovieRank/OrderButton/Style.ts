import styled from 'styled-components';
import { css } from 'styled-components';

const OrderButtonBlock = styled.div`
  width: 83rem;
  height: 2rem;
  margin: auto;
  display: flex;
  justify-content: end;
  margin-bottom: 2rem;
  cursor: pointer;
  @media screen and (max-width: 500px) {
    width: 22rem;
  }
`;
const OrderButtonSubBlock = styled.div`
  border: 3px solid black;
  width: 8rem;
  height: 2rem;
  border-radius: 5px;
  display: flex;
  @media screen and (max-width: 500px) {
    width: 5rem;
  }
`;
const OrderButton = styled.div<{ active: boolean }>`
  display: flex;
  justify-content: center;
  width: 4rem;
  align-items: center;
  font-size: 20px;
  font-weight: bold;

  @media screen and (max-width: 500px) {
    font-size: 12px;
    margin: 0;
    width: 2.5rem;
  }

  ${(props) =>
    props.active &&
    css`
      background: black;
      color: white;
    `}
`;

const styles = {
  OrderButtonBlock,
  OrderButtonSubBlock,
  OrderButton,
};

export default styles;
