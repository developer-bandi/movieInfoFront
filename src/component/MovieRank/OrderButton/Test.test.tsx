import { fireEvent, render, screen } from '@testing-library/react';
import OrderButton from './Presentational';

describe('OrderButton Presentational', () => {
  it('전체 화면 검증', () => {
    const orderchangeMock = jest.fn();
    const utils = render(
      <OrderButton orderchange={orderchangeMock} orderType={'popular'} />
    );
    expect(utils.container).toMatchSnapshot();
    fireEvent.click(screen.getByTestId('popularOrderButton'));
    fireEvent.click(screen.getByTestId('popularOrderButton'));
    expect(orderchangeMock).toBeCalledTimes(2);
  });
});
