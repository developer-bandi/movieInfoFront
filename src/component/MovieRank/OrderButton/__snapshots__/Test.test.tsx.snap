// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderButton <Presentational> 전체 컴포넌트 1`] = `
<div>
  <div
    class="sc-bdvvtL doTKpa"
  >
    <div
      class="sc-gsDKAQ cXqUZX"
    >
      <div
        class="sc-dkPtRN bJNWxh"
        data-testid="popularOrderButton"
      >
        인기순
      </div>
      <div
        class="sc-dkPtRN eDQzrg"
        data-testid="topRatedOrderButton"
      >
        별점순
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderButton Presentational 전체 화면 검증 1`] = `
<div>
  <div
    class="sc-bdvvtL doTKpa"
  >
    <div
      class="sc-gsDKAQ cXqUZX"
    >
      <div
        class="sc-dkPtRN bJNWxh"
        data-testid="popularOrderButton"
      >
        인기순
      </div>
      <div
        class="sc-dkPtRN eDQzrg"
        data-testid="topRatedOrderButton"
      >
        별점순
      </div>
    </div>
  </div>
</div>
`;
